src/script/page/components/FeatureConfigChange/FeatureConfigChangeHandler/FeatureConfigChangeHandler.tsx

```diff
@@ -38,7 +38,7 @@ export function FeatureConfigChangeHandler({teamState}: Props): null {
   useEffect(() => {
     if (config) {
       // initialize feature handlers
-      configureE2EI(logger, config);
+      configureE2EI(logger, config)?.attemptEnrollment();
```

Comment:
This is the actual fix. When we detect the feature change, we trigger the `attemptEnrollment` right away